Index: includes/parser/Parser.php
===================================================================
--- includes/parser/Parser.php	(revision 58867)
+++ includes/parser/Parser.php	(working copy)
@@ -153,7 +153,7 @@
 	 */
 	function __destruct() {
 		if ( isset( $this->mLinkHolders ) ) {
-			$this->mLinkHolders->__destruct();
+			unset( $this->mLinkHolders );
 		}
 		foreach ( $this as $name => $value ) {
 			unset( $this->$name );
