# HG changeset patch
# User root@yourcmc.ru
# Date 1269535930 -10800
Bug 62054 - Remove alpha-bars in categories

diff -r 2f012c39365a -r 7568028ce4c6 extensions/CategoryTree/CategoryPageSubclass.php
--- extensions/CategoryTree/CategoryPageSubclass.php
+++ extensions/CategoryTree/CategoryPageSubclass.php
@@ -141,4 +141,109 @@
 		}
 		parent::finaliseCategoryState();
 	}
+
+    function addPage($title, $sortkey, $pageLength, $isRedirect = false)
+    {
+        $this->titles[] = $title;
+        parent::addPage($title, $sortkey, $pageLength, $isRedirect);
+    }
+
+    function getAllParentCategories($dbr, $title)
+    {
+        $supercats = array($title->getDBkey());
+        $sch = array($title->getDBkey() => true);
+        while ($supercats)
+        {
+            $res = $dbr->select(array('categorylinks', 'page'), 'cl_to', array(
+                'cl_from=page_id',
+                'page_namespace' => NS_CATEGORY,
+                'page_title' => $supercats,
+            ), __METHOD__, array('GROUP BY' => 'cl_to'));
+            $supercats = array();
+            while ($row = $dbr->fetchRow($res))
+            {
+                if (!$sch[$row[0]])
+                {
+                    $supercats[] = $row[0];
+                    $sch[$row[0]] = true;
+                }
+            }
+            $dbr->freeResult($res);
+        }
+        return array_keys($sch);
+    }
+
+    function getPagesSection()
+    {
+        global $wgMinUncatPagesAlphaList;
+        global $wgCategorySubcategorizedList;
+        global $wgOut;
+        if (!$this->articles || !$wgCategorySubcategorizedList && !$wgOut->useSubcategorizedList ||
+            $wgCategorySubcategorizedList && !is_null($wgOut->useSubcategorizedList) &&
+            !$wgOut->useSubcategorizedList)
+            return parent::getPagesSection();
+        $ids = array();
+        foreach ($this->titles as $t)
+            $ids[] = $t->getArticleID();
+        $dbr = wfGetDB(DB_SLAVE);
+        $supercats = $this->getAllParentCategories($dbr, $this->title);
+        $where = array('cl_from' => $ids);
+        foreach ($supercats as $k)
+            $where[] = 'cl_to!='.$dbr->addQuotes($k);
+        $res = $dbr->select('categorylinks', '*', $where, __METHOD__,
+            array('ORDER BY' => 'cl_sortkey'));
+        $cl = array();
+        while ($row = $dbr->fetchRow($res))
+            $cl[$row['cl_to']][] = $row['cl_from'];
+        $dbr->freeResult($res);
+        $new = array();
+        $newkey = array();
+        $done = array();
+        $ids = array_flip($ids);
+        foreach ($cl as $cat => $list)
+        {
+            $cat = str_replace('_', ' ', $cat);
+            foreach ($list as $a)
+            {
+                $new[] = $this->articles[$ids[$a]];
+                $newkey[] = $cat;
+                $done[$ids[$a]] = true;
+            }
+        }
+        $count_undone = 0;
+        for ($i = count($this->articles)-1; $i >= 0; $i--)
+            if (!$done[$i])
+                $count_undone++;
+        $cutoff = $wgMinUncatPagesAlphaList;
+        if (!$cutoff || $cutoff < 0)
+            $cutoff = 10;
+        for ($i = count($this->articles)-1; $i >= 0; $i--)
+        {
+            if (!$done[$i])
+            {
+                array_unshift($new, $this->articles[$i]);
+                if ($count_undone > $cutoff)
+                    array_unshift($newkey, $this->articles_start_char[$i]);
+                else
+                    array_unshift($newkey, $this->title->getText());
+            }
+        }
+        $this->articles = $new;
+        $this->articles_start_char = $newkey;
+        $this->nonplanar_short_list = true;
+        $html = parent::getPagesSection();
+        $this->nonplanar_short_list = false;
+        return $html;
+    }
+
+    function shortList($articles, $articles_start_char)
+    {
+        if ($this->nonplanar_short_list)
+            return parent::shortList($articles, $articles_start_char);
+        $r = '<ul>';
+        foreach ($articles as $a)
+            $r .= "<li>$a</li>";
+        $r .= '</ul>';
+        return $r;
+    }
 }
diff -r 2f012c39365a -r 7568028ce4c6 extensions/CategoryTree/CategoryTree.php
--- extensions/CategoryTree/CategoryTree.php
+++ extensions/CategoryTree/CategoryTree.php
@@ -393,4 +393,32 @@
 	$result = $embed . implode ( "{$pop} {$sep} {$embed}" , $links ) . $pop;
 
 	return false;
-}
\ No newline at end of file
+}
+
+$wgExtensionMessagesFiles['SubcategorizedCategoryPage'] = dirname(__FILE__)."/SubcatCat.i18n.php";
+$wgHooks['MagicWordwgVariableIDs'][] = 'efSubcatCatMagicWordwgVariableIDs';
+$wgHooks['OutputPageParserOutput'][] = 'efSubcatCatOutputPageParserOutput';
+$wgHooks['ParserBeforeInternalParse'][] = 'efSubcatCatParserBeforeInternalParse';
+
+function efSubcatCatMagicWordwgVariableIDs(&$wgVariableIDs)
+{
+    wfLoadExtensionMessages('SubcategorizedCategoryPage');
+    $wgVariableIDs[] = 'nocategorysubcatlist';
+    return true;
+}
+
+function efSubcatCatOutputPageParserOutput(&$out, $parserOutput)
+{
+    if (!is_null($parserOutput->useSubcategorizedList))
+        $out->useSubcategorizedList = $parserOutput->useSubcategorizedList;
+    return true;
+}
+
+function efSubcatCatParserBeforeInternalParse($parser, $text, $stripState)
+{
+    if (MagicWord::get('nocategorysubcatlist')->matchAndRemove($text))
+        $parser->mOutput->useSubcategorizedList = FALSE;
+    if (MagicWord::get('categorysubcatlist')->matchAndRemove($text))
+        $parser->mOutput->useSubcategorizedList = TRUE;
+    return true;
+}
diff -r 2f012c39365a -r 7568028ce4c6 extensions/CategoryTree/SubcatCat.i18n.php
--- /dev/null
+++ extensions/CategoryTree/SubcatCat.i18n.php
@@ -0,0 +1,8 @@
+<?php
+
+$magicWords = array();
+
+$magicWords['en'] = array(
+    'nocategorysubcatlist' => array(0, '__NOCATEGORYSUBCATLIST__'),
+    'categorysubcatlist' => array(0, '__CATEGORYSUBCATLIST__'),
+);
