Index: extensions/Wikilog/Wikilog.php
===================================================================
--- extensions/Wikilog/Wikilog.php	(revision 428)
+++ extensions/Wikilog/Wikilog.php	(working copy)
@@ -361,7 +361,8 @@
 	 */
 	static function getWikilogInfo( $title ) {
 		global $wgWikilogNamespaces;
-
+		if (!$title)
+			return NULL;
 		$ns = MWNamespace::getSubject( $title->getNamespace() );
 		if ( in_array( $ns, $wgWikilogNamespaces ) ) {
 			return new WikilogInfo( $title );
Index: extensions/Wikilog/WikilogUtils.php
===================================================================
--- extensions/Wikilog/WikilogUtils.php    (revision 428)
+++ extensions/Wikilog/WikilogUtils.php    (working copy)
@@ -212,16 +212,9 @@
 			$blocks = preg_split( '/< (h[1-6]) .*? > .*? <\\/\\1>/ix', $content );
 
 			if ( count( $blocks ) > 1 ) {
-				# Long article, get only the first paragraph.
-				$pextr = '/<(p)
-					( \\s+ (?: [^\'"\\/>] | \'[^\']*\' | "[^"]*" )* )?
-					(?: > .*? <\\/\\1\\s*> | \\/> )/isx';
-
-				if ( preg_match_all( $pextr, $blocks[0], $m ) ) {
-					$summary = implode( "\n", $m[0] );
-				} else {
-					$summary = NULL;
-				}
+				$summary = $blocks[0];
+				while (preg_match('#^(.*)<(/?)[a-z0-9\-_:]+[^<>]*>#is', $summary, $m) && !$m[2])
+					$summary = $m[1];
 			} else {
 				# Short article, no summary.
 				$summary = NULL;
