Index: includes/Linker.php
===================================================================
--- includes/Linker.php	(revision 54846)
+++ includes/Linker.php	(working copy)
@@ -244,6 +244,8 @@
 			$query['action'] = 'edit';
 			$query['redlink'] = '1';
 		}
+		if ($target->isTalkPage())
+			$query['section'] = 'new';
 		$ret = $target->getLinkUrl( $query );
 		wfProfileOut( __METHOD__ );
 		return $ret;
