Index: includes/Linker.php
===================================================================
--- includes/Linker.php	(revision 54846)
+++ includes/Linker.php	(working copy)
@@ -244,6 +244,8 @@
 			$query['action'] = 'edit';
 			$query['redlink'] = '1';
 		}
+		if ($target->isTalkPage())
+			$query['section'] = 'new';
 		$ret = $target->getLinkUrl( $query );
 		wfProfileOut( __METHOD__ );
 		return $ret;
Index: includes/SkinTemplate.php
===================================================================
--- includes/SkinTemplate.php	(revision 58867)
+++ includes/SkinTemplate.php	(working copy)
@@ -597,6 +597,8 @@
 		if( $checkEdit && !$title->isKnown() ) {
 			$classes[] = 'new';
 			$query = 'action=edit&redlink=1';
+			if ($title->isTalkPage())
+				$query .= '&section=new';
 		}
 
 		$text = wfMsg( $message );
