Index: includes/FileDeleteForm.php
===================================================================
--- includes/FileDeleteForm.php	(revision 54846)
+++ includes/FileDeleteForm.php	(working copy)
@@ -107,7 +107,7 @@
 			if( $status->ok ) {
 				// Need to delete the associated article
 				$article = new Article( $title );
-				if( wfRunHooks('ArticleDelete', array(&$article, &$wgUser, &$reason)) ) {
+				if( wfRunHooks('ArticleDelete', array(&$article, &$wgUser, $reason, $article->getId())) ) {
 					if( $article->doDeleteArticle( $reason, $suppress ) )
 						wfRunHooks('ArticleDeleteComplete', array(&$article, &$wgUser, $reason));
 				}
Index: extensions/CategoryTree/CategoryTree.php
===================================================================
--- extensions/CategoryTree/CategoryTree.php	(revision 52733)
+++ extensions/CategoryTree/CategoryTree.php	(working copy)
@@ -351,7 +351,7 @@
 * Hook callback that injects messages and things into the <head> tag
 * Does nothing if $parserOutput->mCategoryTreeTag is not set
 */
-function efCategoryTreeParserOutput( &$outputPage, &$parserOutput )  {
+function efCategoryTreeParserOutput( $outputPage, $parserOutput )  {
 	if ( !empty( $parserOutput->mCategoryTreeTag ) ) {
 		CategoryTree::setHeaders( $outputPage );
 	}
@@ -361,7 +361,7 @@
 /**
  * ArticleFromTitle hook, override category page handling
  */
-function efCategoryTreeArticleFromTitle( &$title, &$article ) {
+function efCategoryTreeArticleFromTitle( $title, $article ) {
 	if ( $title->getNamespace() == NS_CATEGORY ) {
 		$article = new CategoryTreeCategoryPage( $title );
 	}
Index: extensions/DocExport/DocExport.php
===================================================================
--- extensions/DocExport/DocExport.php	(revision 327)
+++ extensions/DocExport/DocExport.php	(working copy)
@@ -162,7 +162,7 @@
         echo $html;
     }
 
-    public function onUnknownAction($action, &$article)
+    public function onUnknownAction($action, $article)
     {
         // Here comes all the stuff to show the form and parse it...
         global $wgTitle, $wgOut, $wgUser;
Index: extensions/Wikilog/Wikilog.php
===================================================================
--- extensions/Wikilog/Wikilog.php	(revision 426)
+++ extensions/Wikilog/Wikilog.php	(working copy)
@@ -270,7 +270,7 @@
 		global $wgWikilogNamespaces;
 		if (!$title)
 			return NULL;
-		$ns = Namespace::getSubject( $title->getNamespace() );
+		$ns = MWNamespace::getSubject( $title->getNamespace() );
 		if ( in_array( $ns, $wgWikilogNamespaces ) ) {
 			return new WikilogInfo( $title );
 		} else {
@@ -299,7 +299,7 @@
 	 * @param $title Title object.
 	 */
 	function __construct( $title ) {
-		$ns = Namespace::getSubject( $title->getNamespace() );
+		$ns = MWNamespace::getSubject( $title->getNamespace() );
 		if ( strpos( $title->getText(), '/' ) !== false ) {
 			list( $this->mWikilogName, $this->mItemName ) =
 				explode( '/', $title->getText(), 2 );
Index: extensions/Wikilog/WikilogHooks.php
===================================================================
--- extensions/Wikilog/WikilogHooks.php	(revision 428)
+++ extensions/Wikilog/WikilogHooks.php	(working copy)
@@ -134,9 +134,11 @@
 	 * Purges wikilog metadata when an article is deleted.
 	 * @note This function REQUIRES MediaWiki 1.13 or higher ($id parameter).
 	 */
-	static function ArticleDeleteComplete( &$article, &$user, $reason, $id ) {
+	static function ArticleDeleteComplete( &$article, &$user, $reason, $id = NULL ) {
 		# Retrieve wikilog information.
 		$wi = Wikilog::getWikilogInfo( $article->getTitle() );
+		if ( is_null($id) && $article )
+			$id = $article->getId();
 
 		# Take special procedures if it is a wikilog page.
 		if ( $wi ) {
@@ -253,7 +255,7 @@
 	 * UnknownAction hook handler function.
 	 * Handles action=wikilog requests.
 	 */
-	static function UnknownAction( $action, &$article ) {
+	static function UnknownAction( $action, $article ) {
 		if ( $action == 'wikilog' && $article instanceof WikilogMainPage ) {
 			$article->wikilog();
 			return false;
Index: languages/Language.php
===================================================================
--- languages/Language.php	(revision 47934)
+++ languages/Language.php	(working copy)
@@ -1672,7 +1672,7 @@
 	}
 
 	# Fill a MagicWord object with data from here
-	function getMagic( &$mw ) {
+	function getMagic( $mw ) {
 		if ( !$this->mMagicHookDone ) {
 			$this->mMagicHookDone = true;
 			wfRunHooks( 'LanguageGetMagic', array( &$this->mMagicExtensions, $this->getCode() ) );
