Index: includes/parser/Parser.php
===================================================================
--- includes/parser/Parser.php	(revision 53578)
+++ includes/parser/Parser.php	(working copy)
@@ -3402,7 +3406,7 @@
 	 * @private
 	 */
 	function formatHeadings( $text, $isMain=true ) {
-		global $wgMaxTocLevel, $wgContLang;
+		global $wgMaxTocLevel, $wgContLang, $wgDotAfterTocnumber;
 
 		$doNumberHeadings = $this->mOptions->getNumberHeadings();
 		if( !$this->mTitle->quickUserCan( 'edit' ) ) {
@@ -3544,6 +3548,8 @@
 						$dot = 1;
 					}
 				}
+				if ($wgDotAfterTocnumber)
+					$numbering .= '.';
 			}
 
 			# The safe header is a version of the header text safe to use for links
