Index: includes/specials/SpecialRecentchanges.php
===================================================================
--- includes/specials/SpecialRecentchanges.php	(revision 58867)
+++ includes/specials/SpecialRecentchanges.php	(working copy)
@@ -392,7 +392,7 @@
 
 		$defaults = $opts->getAllValues();
 		$nondefaults = $opts->getChangedValues();
-		$opts->consumeValues( array( 'namespace', 'invert' ) );
+		$opts->consumeValues( array( 'namespace', 'invert', 'categories', 'categories_any' ) );
 
 		$panel = array();
 		$panel[] = $this->optionsPanel( $defaults, $nondefaults );
@@ -529,6 +529,7 @@
 		# Filter articles
 		$articles = array();
 		$a2r = array();
+		$newrows = array();
 		foreach( $rows AS $k => $r ) {
 			$nt = Title::makeTitle( $r->rc_namespace, $r->rc_title );
 			$id = $nt->getArticleID();
@@ -539,8 +540,10 @@
 			if( !isset($a2r[$id]) ) {
 				$a2r[$id] = array();
 			}
+			$newrows[$k] = $r;
 			$a2r[$id][] = $k;
 		}
+		$rows = $newrows;
 
 		# Shortcut?
 		if( !count($articles) || !count($cats) )
