Index: includes/Linker.php
===================================================================
--- includes/Linker.php	(revision 54846)
+++ includes/Linker.php	(working copy)
@@ -369,13 +369,13 @@
 
 		$nt = $this->normaliseSpecialPage( $title );
 
-		if( $nt->getNamespace() == NS_SPECIAL ) {
-			$q = $query;
-		} else if ( '' == $query ) {
-			$q = 'action=edit&redlink=1';
-		} else {
-			$q = 'action=edit&redlink=1&'.$query;
-		}
+		$q = array();
+		if ($nt->isTalkPage())
+			array_push($q, 'section=new');
+		if( $nt->getNamespace() != NS_SPECIAL )
+			array_push($q, 'action=edit', 'redlink=1');
+		array_push($q, $query);
+		$q = implode('&', $q);
 		$u = $nt->escapeLocalURL( $q );
 
 		$titleText = $nt->getPrefixedText();
