Index: languages/messages/MessagesEn.php
===================================================================
--- languages/messages/MessagesEn.php	(revision 58851)
+++ languages/messages/MessagesEn.php	(working copy)
@@ -86,8 +86,8 @@
 	NS_USER_TALK        => 'User_talk',
 	# NS_PROJECT set by $wgMetaNamespace
 	NS_PROJECT_TALK     => '$1_talk',
-	NS_FILE             => 'File',
-	NS_FILE_TALK        => 'File_talk',
+	NS_FILE             => 'Image',
+	NS_FILE_TALK        => 'Image_talk',
 	NS_MEDIAWIKI        => 'MediaWiki',
 	NS_MEDIAWIKI_TALK   => 'MediaWiki_talk',
 	NS_TEMPLATE         => 'Template',
Index: languages/messages/MessagesRu.php
===================================================================
--- languages/messages/MessagesRu.php	(revision 58851)
+++ languages/messages/MessagesRu.php	(working copy)
@@ -51,8 +51,8 @@
 	NS_USER             => 'Участник',
 	NS_USER_TALK        => 'Обсуждение_участника',
 	NS_PROJECT_TALK     => 'Обсуждение_{{GRAMMAR:genitive|$1}}',
-	NS_FILE             => 'Файл',
-	NS_FILE_TALK        => 'Обсуждение_файла',
+	NS_FILE             => 'Изображение',
+	NS_FILE_TALK        => 'Обсуждение_изображения',
 	NS_MEDIAWIKI        => 'MediaWiki',
 	NS_MEDIAWIKI_TALK   => 'Обсуждение_MediaWiki',
 	NS_TEMPLATE         => 'Шаблон',
