Index: extensions/lucene-search/src/org/wikimedia/lsearch/frontend/SearchServer.java
===================================================================
--- extensions/lucene-search/src/org/wikimedia/lsearch/frontend/SearchServer.java	(revision 53812)
+++ extensions/lucene-search/src/org/wikimedia/lsearch/frontend/SearchServer.java	(working copy)
@@ -24,6 +24,7 @@
 package org.wikimedia.lsearch.frontend;
 
 import java.net.ServerSocket;
+import java.net.InetAddress;
 import java.net.Socket;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
@@ -41,6 +42,7 @@
  */
 public class SearchServer extends Thread {
 	private static int port = 8123;
+	private static String host = "0.0.0.0";
 	private static int maxThreads = 80;
 	private static ServerSocket sock;
 	public static String indexPath;
@@ -55,14 +57,15 @@
 	
 	public void startServer(){
 		config = Configuration.open();
-		SearchServer.port = config.getInt("Search","port",8123);
+		SearchServer.port = config.getInt("Search", "port", 8123);
+		SearchServer.host = config.getString("Search", "host", "0.0.0.0");
 		/** Logger */
 		org.apache.log4j.Logger log = Logger.getLogger(SearchServer.class);
 		
 		log.info("Searcher started on port " + port);
 		
 		try {
-			sock = new ServerSocket(port);
+			sock = new ServerSocket(port, 64, InetAddress.getByName(host));
 		} catch (Exception e) {
 			log.fatal("Error: bind error: " + e.getMessage());
 			return;
Index: extensions/lucene-search/src/org/wikimedia/lsearch/frontend/HTTPIndexServer.java
===================================================================
--- extensions/lucene-search/src/org/wikimedia/lsearch/frontend/HTTPIndexServer.java	(revision 53812)
+++ extensions/lucene-search/src/org/wikimedia/lsearch/frontend/HTTPIndexServer.java	(working copy)
@@ -6,6 +6,7 @@
 
 import java.io.IOException;
 import java.net.ServerSocket;
+import java.net.InetAddress;
 import java.net.Socket;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
@@ -38,10 +39,11 @@
 		Configuration config = Configuration.open();
 		org.apache.log4j.Logger log = Logger.getLogger(HTTPIndexServer.class);
 		
-		int port = config.getInt("Index","port",8321);
+		int port = config.getInt("Index", "port", 8321);
+		String host = config.getString("Index", "host", "0.0.0.0");
 		
 		try {
-			sock = new ServerSocket(port);
+			sock = new ServerSocket(port, 64, InetAddress.getByName(host));
 		} catch (Exception e) {
 			log.fatal("Dying: bind error: " + e.getMessage());
 			return;
Index: extensions/lucene-search/template/simple/lsearch.conf
===================================================================
--- extensions/lucene-search/template/simple/lsearch.conf	(revision 53812)
+++ extensions/lucene-search/template/simple/lsearch.conf	(working copy)
@@ -24,6 +24,9 @@
 # Port of http daemon, if different from default 8123
 # Search.port=8000
 
+# Bind address of http daemon (default localhost)
+Search.host=127.0.0.1
+
 # In minutes, how frequently will the index host be checked for updates
 Search.updateinterval=0.1
 
@@ -63,6 +66,9 @@
 # Port of daemon (default is 8321)
 #Index.port=8080
 
+# Bind address of daemon (default localhost)
+Index.host=127.0.0.1
+
 # Maximal queue size after which index is being updated
 Index.maxqueuecount=5000
 
