Index: includes/EditPage.php
===================================================================
--- includes/EditPage.php	(revision 47934)
+++ includes/EditPage.php	(working copy)
@@ -1897,7 +1897,7 @@
 			array(
 				'image'  => $wgLang->getImageFile('button-sig'),
 				'id'     => 'mw-editbutton-signature',
-				'open'   => '--~~~~',
+				'open'   => '~~~~',
 				'close'  => '',
 				'sample' => '',
 				'tip'    => wfMsg('sig_tip'),
