Index: UserMailer.php
===================================================================
--- UserMailer.php	(revision 47934)
+++ UserMailer.php	(working copy)
@@ -101,8 +101,10 @@
 	 * @return mixed True on success, a WikiError object on failure.
 	 */
 	static function send( $to, $from, $subject, $body, $replyto=null ) {
-		global $wgSMTP, $wgOutputEncoding, $wgErrorString, $wgEnotifImpersonal;
+		global $wgSMTP, $wgOutputEncoding, $wgErrorString, $wgEnotifImpersonal, $wgEmailContentType;
 		global $wgEnotifMaxRecips;
+		if (!$wgEmailContentType)
+			$wgEmailContentType = 'text/plain';
 
 		if ( is_array( $to ) ) {
 			wfDebug( __METHOD__.': sending mail to ' . implode( ',', $to ) . "\n" );
@@ -139,7 +141,7 @@
 			$headers['Subject'] = wfQuotedPrintable( $subject );
 			$headers['Date'] = date( 'r' );
 			$headers['MIME-Version'] = '1.0';
-			$headers['Content-type'] = 'text/plain; charset='.$wgOutputEncoding;
+			$headers['Content-type'] = $wgEmailContentType.'; charset='.$wgOutputEncoding;
 			$headers['Content-transfer-encoding'] = '8bit';
 			$headers['Message-ID'] = "<$msgid@" . $wgSMTP['IDHost'] . '>'; // FIXME
 			$headers['X-Mailer'] = 'MediaWiki mailer';
@@ -172,7 +174,7 @@
 			}
 			$headers =
 				"MIME-Version: 1.0$endl" .
-				"Content-type: text/plain; charset={$wgOutputEncoding}$endl" .
+				"Content-type: $wgEmailContentType; charset={$wgOutputEncoding}$endl" .
 				"Content-Transfer-Encoding: 8bit$endl" .
 				"X-Mailer: MediaWiki mailer$endl".
 				'From: ' . $from->toString();
@@ -362,6 +364,10 @@
 					// Already sent an email to this person
 					$userCondition .= ' AND wl_user != ' . intval( $userTalkId );
 				}
+
+				$userCondition .= ' AND wl_notificationtimestamp IS NULL';
+				wfRunHooks('EnotifUserCondition', array(&$this, &$userCondition));
+
 				$dbr = wfGetDB( DB_SLAVE );
 
 				list( $user ) = $dbr->tableNamesN( 'user' );
@@ -373,7 +379,6 @@
 						'wl_title' => $title->getDBkey(),
 						'wl_namespace' => $title->getNamespace(),
 						$userCondition,
-						'wl_notificationtimestamp IS NULL',
 					), __METHOD__ );
 				$userArray = UserArray::newFromResult( $res );
 
@@ -470,6 +475,7 @@
 		$keys['$PAGEMINOREDIT']      = $medit;
 		$keys['$PAGESUMMARY']        = $summary;
 
+		wfRunHooks('EnotifComposeCommonMailtext', array(&$this, &$keys));
 		$subject = strtr( $subject, $keys );
 
 		# Reveal the page editor's address as REPLY-TO address only if
@@ -573,6 +579,8 @@
 			$wgLang->timeanddate( $this->timestamp, true, false, $timecorrection ),
 			$body);
 
+		wfRunHooks('EnotifPersonalizeMailtext', array(&$this, &$watchingUser, &$body));
+
 		return UserMailer::send($to, $this->from, $this->subject, $body, $this->replyto);
 	}
 
Index: EditPage.php
===================================================================
--- EditPage.php	(revision 47934)
+++ EditPage.php	(working copy)
@@ -1897,7 +1897,7 @@
 			array(
 				'image'  => $wgLang->getImageFile('button-sig'),
 				'id'     => 'mw-editbutton-signature',
-				'open'   => '--~~~~',
+				'open'   => '~~~~',
 				'close'  => '',
 				'sample' => '',
 				'tip'    => wfMsg('sig_tip'),
Index: specials/SpecialExport.php
===================================================================
--- specials/SpecialExport.php	(revision 47934)
+++ specials/SpecialExport.php	(working copy)
@@ -212,12 +212,10 @@
 			$pageSet = wfExportGetTemplates( $inputPages, $pageSet );
 		}
 
-		/*
 		// Enable this when we can do something useful exporting/importing image information. :)
 		if( $wgRequest->getCheck( 'images' ) ) {
 			$pageSet = wfExportGetImages( $inputPages, $pageSet );
 		}
-		*/
 
 		$pages = array_keys( $pageSet );
 
@@ -226,6 +224,7 @@
 		$db = wfGetDB( DB_SLAVE );
 		$exporter = new WikiExporter( $db, $history );
 		$exporter->list_authors = $list_authors ;
+		$exporter->dumpUploads = $wgRequest->getCheck('images') ? true : false;
 		$exporter->openStream();
 
 		foreach( $pages as $page ) {
@@ -275,7 +274,7 @@
 	}
 	$form .= Xml::checkLabel( wfMsg( 'export-templates' ), 'templates', 'wpExportTemplates', false ) . '<br />';
 	// Enable this when we can do something useful exporting/importing image information. :)
-	//$form .= Xml::checkLabel( wfMsg( 'export-images' ), 'images', 'wpExportImages', false ) . '<br />';
+	$form .= Xml::checkLabel( wfMsg( 'export-images' ), 'images', 'wpExportImages', false ) . '<br />';
 	$form .= Xml::checkLabel( wfMsg( 'export-download' ), 'wpDownload', 'wpDownload', true ) . '<br />';
 
 	$form .= Xml::submitButton( wfMsg( 'export-submit' ) );
Index: specials/SpecialUpload.php
===================================================================
--- specials/SpecialUpload.php	(revision 47934)
+++ specials/SpecialUpload.php	(working copy)
@@ -4,6 +4,9 @@
  * @ingroup SpecialPage
  */
 
+# A CRUTCH fixing 'Fatal error: Cannot redeclare wfspecialupload() (previously declared in /usr/share/wikis/wiki/includes/specials/SpecialUpload.php:12) in /usr/share/wikis/wiki/includes/specials/SpecialUpload.php on line 15'
+if (!function_exists('wfSpecialUpload'))
+{
 
 /**
  * Entry point
@@ -1753,3 +1756,5 @@
 		}
 	}
 }
+
+}
