# HG changeset patch
# User root@yourcmc.ru
# Date 1269560994 -10800
Bug 61871 - Allow (optional) article merging for SAME user

diff -r 2f012c39365a -r 46bd18b9b11f includes/EditPage.php
--- includes/EditPage.php
+++ includes/EditPage.php
@@ -730,6 +730,7 @@
 	function internalAttemptSave( &$result, $bot = false ) {
 		global $wgFilterCallback, $wgUser, $wgOut, $wgParser;
 		global $wgMaxArticleSize;
+		global $wgSuppressSameUserConflicts;
 
 		$fname = 'EditPage::attemptSave';
 		wfProfileIn( $fname );
@@ -910,7 +911,7 @@
 		}
 
 		# Suppress edit conflict with self, except for section edits where merging is required.
-		if ( $this->section == '' && $userid && $this->userWasLastToEdit($userid,$this->edittime) ) {
+		if ( $wgSuppressSameUserConflicts && $this->section == '' && $userid && $this->userWasLastToEdit($userid,$this->edittime) ) {
 			wfDebug( "EditPage::editForm Suppressing edit conflict, same user.\n" );
 			$this->isConflict = false;
 		} else {
