Index: extensions/MediaFunctions/MediaFunctions.php
===================================================================
--- extensions/MediaFunctions/MediaFunctions.php	(revision 59371)
+++ extensions/MediaFunctions/MediaFunctions.php	(working copy)
@@ -30,6 +30,7 @@
  */
 function efMediaFunctionsSetup() {
 	global $wgParser;
+	wfLoadExtensionMessages('MediaFunctions');
 	$wgParser->setFunctionHook( 'mediamime', array( 'MediaFunctions', 'mediamime' ) );
 	$wgParser->setFunctionHook( 'mediasize', array( 'MediaFunctions', 'mediasize' ) );
 	$wgParser->setFunctionHook( 'mediaheight', array( 'MediaFunctions', 'mediaheight' ) );
